integrationsupport/leafhasher.go

```diff
@@ -0,0 +1,31 @@
+package integrationsupport
+
+import (
+	"github.com/datatrails/go-datatrails-simplehash/simplehash"
+)
+
+// LeafType provides domain separation for the different kinds of tree leaves we require.
+type LeafType uint8
+
+const (
+	// LeafTypePlain is used for committing to plain values.
+	LeafTypePlain LeafType = iota
+	// LeafTypePeriodSentinel is entered into the MMR once per period. By
+	// forcing a heartbeat entry, we guarantee a liveness indicator - their will
+	// be a definable lower bound on how often the MMR root changes
+	LeafTypePeriodSentinel
+	// LeafTypeEpochTombstone is always the last leave in an epoch MMR. This is
+	// used to provide crash fault tolerance on the epoch as whole
+	LeafTypeEpochTombStone
+)
+
+type LeafHasher struct {
```

Comment:
i think we already have a hasher defined in logverification?

`NewLogVersion0Hasher().HashEvent(eventJson)`

Is it possible to use that hasher?

Reply:
fixforward

Reply:
Will leave in, as hashing method redundancy can help detect drift